rust/sdk/examples/chat/display.rs

```diff
@@ -127,6 +104,200 @@ pub async fn display(mut cell: ActoCell<Display, impl ActoRuntime>) {
     }
 }
 
+fn render<W: io::Write>(
+    f: &mut Frame<CrosstermBackend<W>>,
+    identity: &Reader<Identity>,
+    text_area: &Reader<TextArea>,
+    messages: &Reader<Vec<Message>>,
+    scroll_state: &mut MessageScrollState,
+    not_connected: &Option<String>,
+) {
+    let is_editing = identity.project(|identity| identity.edit.is_some());
+    if is_editing {
+        render_editing_identity(f, identity);
+    } else {
+        render_chat(f, text_area, messages, scroll_state, not_connected);
+    }
+}
+
+fn render_editing_identity<W: io::Write>(f: &mut Frame<CrosstermBackend<W>>, identity: &Reader<Identity>) {
+    let size = f.size();
+    let pad = (size.height / 2).checked_sub(2).unwrap_or(0);
+    let layout = Layout::default()
+        .direction(ratatui::prelude::Direction::Vertical)
+        .constraints([
+            ratatui::prelude::Constraint::Length(pad),
+            ratatui::prelude::Constraint::Max(3),
```

Comment:
This should probably be `Length(3)` since a terminal of less than three lines won’t work anyway.

Reply:
Ah that is right.

Reply:
fixed in https://github.com/Actyx/Actyx/pull/554/commits/a52d3233b2e80053dbe04d872393d0b23e2aa816